### PR TITLE
Add property to string that checks wether it's a valid email

### PR DESCRIPTION
**Pull request by marijnschilling**

### CODE REVIEW

File: Tests/Tests.swift [hunk 1]
```diff
@@ -18,4 +18,12 @@ class Tests: XCTestCase {
         almostRight = d <= 90.1 && d >= 89.99
         XCTAssert(almostRight)
     }
+
+    func testValidEmail() {
```
Comment (elland) [inline Tests/Tests.swift:22]: Can you add all the tests back? :)

File: Sources/String+Sweetness.swift [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import Foundation
+
+public extension String {
+    public var isValidEmail: Bool {
+        let emailRegEx = ".*@.*"
```
Comment (3lvis) [inline Sources/String+Sweetness.swift:5]: I liked what you had before.

Comment (elland) [reply Sources/String+Sweetness.swift:5]: Didn't we discuss this all 3 of us yesterday?

Comment (3lvis) [reply Sources/String+Sweetness.swift:5]: I didn't have time to give input, I was busy. 😞

Comment (elland) [reply Sources/String+Sweetness.swift:5]: Go for it. What would be your argument for the other regex?

Comment (3lvis) [reply Sources/String+Sweetness.swift:5]: It doesn't have to be the previous one, but at least it should check for the current cases -> https://github.com/UseSweet/SweetFoundation/pull/16

Either that or it should check for whatever the backend accepts, if we know what it accepts.

Comment (elland) [reply Sources/String+Sweetness.swift:5]: Backend should accept any valid email. Only regex that matches all valid emails that I know of is the top one. If you've come across one that implements the whole email RFC, please link. 😄

Comment (3lvis) [reply Sources/String+Sweetness.swift:5]: It doesn't, though. This is not valid `@invalidemail.nl`.

Comment (elland) [reply Sources/String+Sweetness.swift:5]: I think I'm fine with `.+@.+\..{2,}`. 🤔

Comment (marijnschilling) [reply Sources/String+Sweetness.swift:5]: I checked with isabella she will set the backend to make it ".*@.*" as well..

Comment (3lvis) [reply Sources/String+Sweetness.swift:5]: I would like if we had this here anyway. Something that checks

`[validcharacters]`@`[validcharacters]`.`[validcharacters]`

`validcharacters` of course won't contain @ or .

Comment (3lvis) [reply Sources/String+Sweetness.swift:5]: It doesn't have to be a regex, btw.

Comment (elland) [reply Sources/String+Sweetness.swift:5]: @3lvis overkill. No one is trying to intentionally register broken invalid emails, and if they are, so what? 🤔

Comment (3lvis) [reply Sources/String+Sweetness.swift:5]: @Elland That's another discussion, then you would have to check why we need to check valid emails at all. What I was referring to is that if we're doing it, better to do it well.

Comment (elland) [reply Sources/String+Sweetness.swift:5]: @3lvis did you read the article I sent yesterday?

Comment (3lvis) [reply Sources/String+Sweetness.swift:5]: No, I don't remember seeing it.

**Comment by designatednerd** [comment]
@marijnschilling Are you interested in continuing this experiment or nah?

**Comment by designatednerd** [comment]
One thing I would point out is a former colleague of mine has done some *extensive* testing around this, and while you can generally do OK with [W3C's regex](https://www.w3.org/TR/2012/WD-html-markup-20120320/input.email.html) the only real thing that never gives false negatives is checking if there's an "@" in the string, and nothing more. 

[This is what he wound up doing for emails](https://github.com/vokal/VOKUtilities/blob/master/Pod/NSString%2BVOKValidation/NSString%2BVOKValidation.m#L53), fwiw.

**Comment by elland** [comment]
@designatednerd that solution looks good to me. Would only add a check that the @ symbol is not at the end of the string either. 😂

**Comment by designatednerd** [comment]
ok. @marijnschilling do you want to pick up on this, or should I take this over?

**Comment by elland** [comment]
Can we close this? 👍

**Comment by designatednerd** [comment]
I'm fine with that. @marijnschilling, @3lvis?

**Comment by designatednerd** [comment]
I'm gonna go ahead and close this out - we should revisit this at some point with the perils of email validation in mind

**Pull request closed without merging**